./src/job_description_feature_extraction.py:8:84: W291 trailing whitespace
./src/job_description_feature_extraction.py:12:29: E231 missing whitespace after ','
./src/job_description_feature_extraction.py:12:40: E231 missing whitespace after ','
./src/job_description_feature_extraction.py:12:52: E231 missing whitespace after ','
./src/job_description_feature_extraction.py:12:61: E231 missing whitespace after ','
./src/job_description_feature_extraction.py:12:71: W291 trailing whitespace
./src/job_description_feature_extraction.py:13:21: E225 missing whitespace around operator
./src/job_description_feature_extraction.py:17:42: E261 at least two spaces before inline comment
./src/job_description_feature_extraction.py:18:33: E261 at least two spaces before inline comment
./src/job_description_feature_extraction.py:22:1: W293 blank line contains whitespace
./src/job_description_feature_extraction.py:26:1: E302 expected 2 blank lines, found 1
./src/job_description_feature_extraction.py:30:83: E261 at least two spaces before inline comment
./src/job_description_feature_extraction.py:31:1: W293 blank line contains whitespace
./src/job_description_feature_extraction.py:33:33: E221 multiple spaces before operator
./src/job_description_feature_extraction.py:33:101: E501 line too long (124 > 100 characters)
./src/job_description_feature_extraction.py:33:104: E261 at least two spaces before inline comment
./src/job_description_feature_extraction.py:34:68: E261 at least two spaces before inline comment
./src/job_description_feature_extraction.py:35:1: W293 blank line contains whitespace
./src/job_description_feature_extraction.py:36:56: E261 at least two spaces before inline comment
./src/job_description_feature_extraction.py:37:47: E261 at least two spaces before inline comment
./src/job_description_feature_extraction.py:38:1: W293 blank line contains whitespace
./src/job_description_feature_extraction.py:40:35: E261 at least two spaces before inline comment
./src/job_description_feature_extraction.py:42:18: E261 at least two spaces before inline comment
./src/job_description_feature_extraction.py:43:33: E221 multiple spaces before operator
./src/job_description_feature_extraction.py:44:1: W293 blank line contains whitespace
./src/job_description_feature_extraction.py:46:56: E261 at least two spaces before inline comment
./src/job_description_feature_extraction.py:47:33: W291 trailing whitespace
./src/main.py:13:77: W291 trailing whitespace
./src/main.py:14:77: W291 trailing whitespace
./src/main.py:16:88: W292 no newline at end of file

wow, was it this strict :scream:

--- a/src/job_description_feature_extraction.py
+++ b/src/job_description_feature_extraction.py
@@ -9,8 +9,8 @@ def get_one_hot_encoded_words(feature_to_extract):
     From those lists, the words appear in the list below ( word_list ) was hand picked
     Fhey are the word which might have an impact on the salary
     """
-    word_list = ['excellent','graduate','immediate','junior','urgent']   
-    encoded_features= []
+    word_list = ['excellent', 'graduate', 'immediate', 'junior', 'urgent']
+    encoded_features = []
     for item in word_list:
         one_hot_encoded_list = []
         for data in feature_to_extract.iterrows():
@@ -19,31 +19,32 @@ def get_one_hot_encoded_words(feature_to_extract):
                 one_hot_encoded_list.append(1)
             else:
                 one_hot_encoded_list.append(0)
-            
         encoded_features.append(one_hot_encoded_list)
     return encoded_features
+
 
 def get_feature_word_count(column_to_count):
     count_dict = {}
     for data in column_to_count.iterrows():
         string_data = str(data)
-        data_digits_removed = ''.join([i for i in string_data if not i.isdigit()])# Remove digits
+        data_digits_removed = ''.join([i for i in string_data if not i.isdigit()])  # Remove digits
         
         puct_exclude = set(string.punctuation)
-        data_puctuations_removed  = ''.join(ch for ch in data_digits_removed if ch not in puct_exclude)# Remove punctuations
-        data_puctuations_removed = data_puctuations_removed.lower()# Lowercase the string
+        data_puctuations_removed  = ''.join(ch for ch in data_digits_removed if ch not in puct_exclude)
+        # Remove punctuations
+        data_puctuations_removed = data_puctuations_removed.lower()  # Lowercase the string
         
-        data_as_list = data_puctuations_removed.split()# Turn into list
-        counts_as_dict = Counter(data_as_list)# Turn into dictionary with counts
+        data_as_list = data_puctuations_removed.split()  # Turn into list
+        counts_as_dict = Counter(data_as_list)  # Turn into dictionary with counts
         
         for item in counts_as_dict:
-            if item in count_dict:# If item exists in the dict
+            if item in count_dict:  # If item exists in the dict
                 count_dict[item] = count_dict[item]+1
-            else:# If item doesnt exists
-                count_dict[item]  = 1
+            else:  # If item doesnt exists
+                count_dict[item] = 1
         
     for item in list(count_dict):
-        if count_dict[item] < len(column_to_count)/200:# Anything below 0.5 is insignificant
+        if count_dict[item] < len(column_to_count)/200:  # Anything below 0.5 is insignificant
             del count_dict[item]  
     count_dict = sorted(count_dict.items(), key=operator.itemgetter(0))
     # Returns a list of tuples which has word:count

--- a/src/job_description_feature_extraction.py
+++ b/src/job_description_feature_extraction.py
@@ -33,7 +33,7 @@ def get_feature_word_count(column_to_count):
         puct_exclude = set(string.punctuation)
         data_punct_removed = ''.join(ch for ch in data_digits_removed if ch not in puct_exclude)
         # Remove punctuations
-        data_punct_removed = data_puctuations_removed.lower()  # Lowercase the string
+        data_punct_removed = data_punct_removed.lower()  # Lowercase the string
         data_as_list = data_punct_removed.split()  # Turn into list
         counts_as_dict = Counter(data_as_list)  # Turn into dictionary with counts
         for item in counts_as_dict:

--- a/src/job_description_feature_extraction.py
+++ b/src/job_description_feature_extraction.py
@@ -5,7 +5,7 @@ from collections import Counter
 
 def get_one_hot_encoded_words(feature_to_extract):
     """
-    get_feature_word_count(x) returns a list of tuples which consists of word:count 
+    get_feature_word_count(x) returns a list of tuples which consists of word:count
     From those lists, the words appear in the list below ( word_list ) was hand picked
     Fhey are the word which might have an impact on the salary
     """
@@ -14,8 +14,8 @@ def get_one_hot_encoded_words(feature_to_extract):
     for item in word_list:
         one_hot_encoded_list = []
         for data in feature_to_extract.iterrows():
-            sentence = data[1][0].lower()# Lower casing the sentence
-            if item in sentence: #
+            sentence = data[1][0].lower()  # Lower casing the sentence
+            if item in sentence:
                 one_hot_encoded_list.append(1)
             else:
                 one_hot_encoded_list.append(0)
@@ -27,25 +27,23 @@ def get_feature_word_count(column_to_count):
     count_dict = {}
     for data in column_to_count.iterrows():
         string_data = str(data)
-        data_digits_removed = ''.join([i for i in string_data if not i.isdigit()])  # Remove digits
-        
+        data_digits_removed = ''.join(
+            [i for i in string_data if not i.isdigit()]
+        )  # Remove digits
         puct_exclude = set(string.punctuation)
-        data_puctuations_removed  = ''.join(ch for ch in data_digits_removed if ch not in puct_exclude)
+        data_punct_removed = ''.join(ch for ch in data_digits_removed if ch not in puct_exclude)
         # Remove punctuations
-        data_puctuations_removed = data_puctuations_removed.lower()  # Lowercase the string
-        
-        data_as_list = data_puctuations_removed.split()  # Turn into list
+        data_punct_removed = data_puctuations_removed.lower()  # Lowercase the string
+        data_as_list = data_punct_removed.split()  # Turn into list
         counts_as_dict = Counter(data_as_list)  # Turn into dictionary with counts
-        
         for item in counts_as_dict:
             if item in count_dict:  # If item exists in the dict
                 count_dict[item] = count_dict[item]+1
             else:  # If item doesnt exists
                 count_dict[item] = 1
-        
     for item in list(count_dict):
         if count_dict[item] < len(column_to_count)/200:  # Anything below 0.5 is insignificant
-            del count_dict[item]  
+            del count_dict[item]
     count_dict = sorted(count_dict.items(), key=operator.itemgetter(0))
     # Returns a list of tuples which has word:count
     return count_dict

--- a/src/main.py
+++ b/src/main.py
@@ -10,7 +10,7 @@ if __name__ == "__main__":
     raw_data = pd.read_csv(raw_data_csv_file_name)
     description_feature = raw_data[['FullDescription']]
     one_hot_encoded_features = get_one_hot_encoded_words(description_feature)
-    # one_hot_encoded_features are a list of 5 lists consisting of 1s and 0s 
-    # If a word appears in the description, this kind of representation will 
+    # one_hot_encoded_features are a list of 5 lists consisting of 1s and 0s
+    # If a word appears in the description, this kind of representation will
     # make it feasible for machine learning
     # We will need to append those lists on final data frame when everything is cleaned

